core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/upgrade/extensions/replication/repmgr-for-pg-10-upgrade.yml

```diff
@@ -0,0 +1,136 @@
+---
+# Documentation: https://repmgr.org/docs/5.2/upgrading-repmgr-extension.html#UPGRADING-MAJOR-VERSION#
+# Compatibility: https://repmgr.org/docs/5.2/install-requirements.html#INSTALL-COMPATIBILITY-MATRIX
+
+# Expects:
+#   - pg_service_names [dict] (list for each OS family)
+
+- name: repmgr for PG 10 | Create upgrade state file
+  file:
+    path: "{{ upgrade.repmgr.upgrade_state_file_path }}"
+    state: touch
+    mode: u=rw,g=r,o=
+
+- name: repmgr for PG 10 | Load postgresql role manifest
+  include_vars:
+    file: manifest.yml
+    name: postgresql_manifest
+
+- name: repmgr for PG 10 | Search for primary node
+  become_user: postgres
+  # command prints primary node name (hostname)
+  shell: |-
+    set -o pipefail && \
+    {{ pg_10.repmgr.bin_dir[ansible_os_family] }}/repmgr cluster show \
+      | awk 'BEGIN{FS="|"} {gsub(/ /,""); if ($3 == "primary") print $2}'
+  changed_when: false
+  register: find_pg_primary_node
+  failed_when: find_pg_primary_node.rc != 0 or find_pg_primary_node.stdout == ""
+  run_once: true
+  args:
+    executable: /bin/bash
+
+# Step: Stop repmgr service
+- name: repmgr for PG 10 | Stop repmgr service
+  systemd:
+    name: "{{ pg_10.repmgr.service_name[ansible_os_family] }}"
+    state: stopped
+
+# Step: Disable repmgr service to prevent packages from prematurely restarting repmgr
+- name: repmgr for PG 10 | Disable repmgr service
+  systemd:
+    name: "{{ pg_10.repmgr.service_name[ansible_os_family] }}"
+    enabled: false
+
+# Step: Install repmgr packages
+
+# On Ubuntu there is dependent 'repmgr-common' package.
+# apt module doesn't support --allow-downgrades (see https://github.com/ansible/ansible/issues/29451)
+# so we keep installed version if it's newer.
+- name: repmgr for PG 10 | Set target version for repmgr-common package
+  set_fact:
+    repmgr_common_target_version: >-
+      {{ _installed_version is version(target_version, '>') | ternary(_installed_version, target_version + '-*') }}
+  when:
+    - ansible_os_family == 'Debian'
+    - ansible_facts.packages['repmgr-common'] is defined
+  vars:
+    _installed_version: "{{ ansible_facts.packages['repmgr-common'][0].version }}"
+
+- name: repmgr for PG 10 | Install repmgr package(s)
+  package:
+    name: "{{ _packages[ansible_os_family] }}"
+    state: present
+  vars:
+    _packages:
+      Debian:
+        - "{{ repmgr_package_name }}={{ target_version + '-*' }}"
+        - repmgr-common={{ repmgr_common_target_version | default(target_version + '-*') }}
+      RedHat:
+        - "{{ repmgr_package_name }}-{{ target_version }}"
+  module_defaults:
+    yum: { lock_timeout: "{{ yum_lock_timeout }}" }
+
+# Step: systemctl daemon-reload
+# Step: Restart PostgreSQL
+- name: repmgr for PG 10 | Restart PostgreSQL service
+  systemd:
+    name: "{{ service_name }}"
+    state: restarted
+    daemon_reload: true
+  loop_control:
+    loop_var: service_name
+  loop: "{{ pg_service_names[ansible_os_family] }}"
```

Comment:
```
  loop_control:
    loop_var: service_name
  loop:
    - "{{ pg_10.pg.service_name['ansible_os_family'] }}"
    - "{{ pg_10.pg.instantiated_service_name['ansible_os_family'] }}"
```

Reply:
I've split the restart task into 2 in order to avoid double restart of the child (instantiated) service since it is restarted by the parent (main) service. The second (added) task ensures the child service is running after the restart.